ui_framework/components/modal/modal.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+export function KuiModal({ children, ...rest }) {
+  return (
+    <div className="kuiModal" { ...rest }>
```

Comment:
I agree with you @w33ble, but we've had this discussion in a previous PR for React buttons. Here's some context:

* Original comment: https://github.com/elastic/kibana/pull/10646#discussion_r107442633
* Discussion: https://github.com/elastic/kibana/pull/10646#issuecomment-288511514

That being said, you're right about the problem with clobbering the `className`. We should be concatenating these strings, not overriding them:

```javascript
const {
  className,
  children,
  ...rest
} = props

const classes = classnames('my-class', className)

return <a className={ classes } { ...rest }>{ children }</a>
```